port/unix/omrsock.c

```diff
@@ -698,19 +765,51 @@ omrsock_recvfrom(struct OMRPortLibrary *portLibrary, omrsock_socket_t sock, uint
 int32_t
 omrsock_pollfd_init(struct OMRPortLibrary *portLibrary, omrsock_pollfd_t handle, omrsock_socket_t sock, int16_t events)
 {
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+	if (NULL == handle) {
+		return OMRPORT_ERROR_INVALID_ARGUMENTS;
+	}
+	memset(handle, 0, sizeof(OMRPollFd));
+	handle->socket = sock;
+	handle->events = get_omr_poll_constant(events);
+	return 0;
 }
 
 int32_t
 omrsock_get_pollfd_info(struct OMRPortLibrary *portLibrary, omrsock_pollfd_t handle, omrsock_socket_t *sock, int16_t *revents)
 {
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+	if (NULL == handle) {
+		return OMRPORT_ERROR_INVALID_ARGUMENTS;
+	}
+	*sock = handle->socket;
+	*revents = get_os_poll_constant(handle->revents);
 }
 
 int32_t
 omrsock_poll(struct OMRPortLibrary *portLibrary, omrsock_pollfd_t fds, uint32_t nfds, int32_t timeoutMs)
 {
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+	int32_t i = 0;
+	int32_t numPollFdSet = 0;
+	struct pollfd pfds[nfds];
```

Comment:
nfds is a dynamic value, and we can't stack-allocate a dynamic number of pollfd's.

Stack allocate a small fixed number of pollfds. Use this array if the number of fd's fits in this array. Otherwise, dynamically allocate (eg malloc) an array of pollfd's of the correct size.